.github/copilot-instructions.md

```diff
@@ -0,0 +1,289 @@
+# Skyrim Community Shaders
+
+**ALWAYS follow these instructions first and only fallback to additional search and context gathering if the information is incomplete or found to be in error.**
+
+SKSE64 plugin providing modular DirectX 11 graphics enhancements for Skyrim SE/AE/VR. Features runtime shader compilation, 25+ graphics features, and cross-platform Skyrim variant support.
+
+## Critical Build Requirements
+
+**This is a Windows-only project requiring specific toolchain setup:**
+
+- Visual Studio Community 2022 with "Desktop development with C++" workload
+- CMake 3.21+ (add to PATH environment variable)
+- Git (add to PATH environment variable)  
+- vcpkg with VCPKG_ROOT environment variable set
+- Windows SDK
+- **NEVER CANCEL BUILDS**: Full builds take 45-60 minutes, shader validation takes 15-30 minutes
+
+## Environment Limitations
+
+**In Linux/WSL environments:**
+- **Cannot build natively** - Windows-only with Visual Studio dependency
+- **Cannot validate shaders** - requires Windows fxc.exe DirectX shader compiler  
+- **Docker limitation** - requires Windows Docker containers (Windows host only)
+- **Repository tasks work** - can clone, examine code, run Python analysis tools
+- **Use case**: Code review, documentation, Python tooling only
+
+## Working Effectively
+
+### Initial Repository Setup
+```bash
+# Clone with submodules (CRITICAL - takes 2-5 minutes)
+git clone https://github.com/doodlum/skyrim-community-shaders.git --recursive
+cd skyrim-community-shaders
+
+# If cloned without --recursive, initialize submodules
+git submodule update --init --recursive
+```
+
+### Build Commands (Windows Only)
+```powershell
+# Primary build command - NEVER CANCEL: Takes 45-60 minutes
+./BuildRelease.bat [PRESET]
+
+# Available presets:
+# ALL (default) - Universal binary for SE/AE/VR
```

Comment:
Development is almost always exclusively with all. The other presets are legacy.